test/cpp_extensions/open_registration_extension/torch_openreg/csrc/runtime/OpenRegDeviceAllocator.cpp

```diff
@@ -1,8 +1,205 @@
 #include "OpenRegDeviceAllocator.h"
 
+#include <c10/util/Exception.h>
+#include <c10/util/irange.h>
+
 namespace c10::openreg {
 
-static OpenRegDeviceAllocator global_openreg_alloc;
-REGISTER_ALLOCATOR(c10::DeviceType::PrivateUse1, &global_openreg_alloc);
+// Anonymous namespace for internal implementation
+namespace {
+
+// Static global allocator instance
+static OpenRegDeviceAllocator allocator;
```

Comment:
```suggestion
OpenRegDeviceAllocator g_allocator;
```

- We do not need to add static prefix in anonymous namespace, right?
- It's best to add the prefix `g_` before global variables to indicate that they are global variables.